mypy/semanal.py

```diff
@@ -4381,3 +4410,21 @@ def find_fixed_callable_return(expr: Expression) -> Optional[CallableType]:
             if isinstance(t.ret_type, CallableType):
                 return t.ret_type
     return None
+
+
+def mark_any_non_explicit(t: Type) -> None:
+    """For all types within t, if that type is Any, set explicit to False"""
+    t.accept(MarkAnyNonExplicit())
+
+
+class MarkAnyNonExplicit(TypeQuery[None]):
```

Comment:
What about having a `TypeQuery` that returns all `Any` types that are part of a type, and then iterate over the results of the query and set `explicit`? I.e. something like this:

```py
for any_type in find_all_any_types(t):
    any_type.explicit = False
```